js/views/modals/purchase/ActionBtn.js

```diff
@@ -0,0 +1,93 @@
+import $ from 'jquery';
+import loadTemplate from '../../../utils/loadTemplate';
+import baseView from '../../baseVw';
+import Listing from '../../../models/listing/Listing';
+
+export default class extends baseView {
+  constructor(options = {}) {
+    if (!options.listing || !(options.listing instanceof Listing)) {
+      throw new Error('Please provide a listing model.');
+    }
+    if (!options.state) {
+      throw new Error('Please provide a state object.');
+    }
+
+    super(options);
+    this.options = options;
+    this.state = options.state;
+
+    $(document).on('click', this.onDocumentClick.bind(this));
+  }
+
+  className() {
+    return 'actionBtn';
+  }
+
+  events() {
+    return {
+      'click .js-payBtn': 'clickPayBtn',
+      'click .js-pendingBtn': 'clickPendingBtn',
+      'click .js-confirmPayConfirm': 'clickConfirmBtn',
+      'click .js-confirmPayCancel': 'closeConfirmPay',
+    };
+  }
+
+  onDocumentClick(e) {
+    if (this.state.phase === 'confirm' && !($.contains(this.$confirmPay[0], e.target))) {
+      this.state.phase = 'pay';
+      this.render();
+    }
+  }
+
+  clickPayBtn(e) {
+    e.stopPropagation();
+    this.state.phase = 'confirm';
+    this.render();
+  }
+
+  clickConfirmBtn() {
+    this.state.phase = 'pay';
+    this.$payBtn.addClass('processing');
+    this.$confirmPay.addClass('hide');
+    this.trigger('purchase');
+  }
+
+  closeConfirmPay() {
+    this.state.phase = 'pay';
+    this.render();
+  }
+
+  clickPendingBtn() {
+    this.state.phase = 'close';
+    this.render();
+  }
+
+  get $payBtn() {
+    return this._$payBtn ||
+      (this._$payBtn = this.$('.js-payBtn'));
+  }
+
+  get $confirmPay() {
+    return this._$confirmPay ||
+      (this._$confirmPay = this.$('.js-confirmPay'));
+  }
+
+  remove() {
+    $(document).off('click', this.onDocumentClick);
```

Comment:
This is a bug I need to also fix in Configuration.js (where you got the code), but bind() returns a new function and that is actually what need to be unbound:

// in the constructor
this.boundOnDocClick = this.onDocumentClick.bind(this);
$(document).on('click', this.boundOnDocClick);

// in remove
$(document).off('click', this.boundOnDocClick);

otherwise this doc click handler is just going to perpetually live and a new one will keep getting applied (and keep on living) every time the Purchase modal is opened.